File: Data Structures & Algorithms/is-anagram/submission-2.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        bool res = true;

        unordered_map<char, int> st;

        if(s.size() != t.size()) {
            res = false;
            return res;
        }

        for(int i = 0; i < (int)s.size(); i ++) {
            st[s[i]] += 1;
        }

        for(int i=0; i < (int)t.size(); i ++) {
            if(st.count(t[i]) > 0 and st[t[i]] > 0) {
                st[t[i]] -= 1;
            }
            else {
                res = false;
                break;
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/gas-station/submission-2.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        temp = []

        for i in range(len(gas)):
            temp.append(gas[i] - cost[i])
        
        res = -1
        max_till_now = float('-inf')

        if(sum(temp) >= 0):
            #A solution exists
            for i in range(len(temp)):
                if(temp[i] >= 0):
                    cur = 0
                    j = i
                    count = 0
                    while(count < len(temp)):
                        cur += temp[j]
                        if(cur < 0):
                            break
                        j += 1
                        j = j % len(temp)
                        count += 1
                    
                    if(cur >= 0):
                        res = i
                        break


        return res